docs/proposals/gpu-support.md

```diff
@@ -0,0 +1,305 @@
+
+    $ curl -s localhost:3476/docker/cli
+    --device=/dev/nvidiactl --device=/dev/nvidia-uvm --device=/dev/nvidia0
+    --device=/dev/nvidia1 --volume-driver=nvidia-docker
+    --volume=nvidia_driver_352.68:/usr/local/nvidia:ro
+
+An obvious change is an additional endpoint for JSON output. The unresolved
+question is whether the plugin should be run standalone like that (called over
+HTTP, perhaps with endpoints for a new Kubernetes resource API) or whether the
+relevant code from its `nvidia` package should be linked directly into kubelet.
+A partial list of tradeoffs:
+
+|                     | External binary                                                                                   | Linked in                                                    |
+|---------------------|---------------------------------------------------------------------------------------------------|--------------------------------------------------------------|
+| Use of cgo          | Confined to binary                                                                                | Linked into kubelet, but with lazy binding                   |
+| Expandibility       | Limited if we run the plugin, increased if library is used to build a Kubernetes-tailored daemon. | Can reuse the `nvidia` library as we prefer                  |
+| Bloat               | None                                                                                              | Larger kubelet, even for systems without GPUs                |
+| Reliability         | Need to handle the binary disappearing at any time                                                | Fewer headeaches                                             |
+| (Un)Marshalling     | Need to talk over JSON                                                                            | None                                                         |
+| Administration cost | One more daemon to install, configure and monitor                                                 | No extra work required, other than perhaps configuring flags |
+| Releases            | Potentially on its own schedule                                                                   | Tied to Kubernetes'                                          |
+
+For all its drawbacks, the external binary solution is probably a better fit for
+V0.
+
+However we obtain the data, we'll need to change Kubelet's `dockertools` to also
+map devices into the containers and manage the CUDA volumes. As shown above,
+volumes are named after the hardware driver version.
+
+## Timeline
+
+### V0 (targeting Kubernetes 1.3)
+
+The first two tracks can progress in parallel.
+
+#### Scheduling
+
+1. Write scheduler extender, assuming one GPU per machine and user-supplied node
+labels for resource inventory
+1. Support fractional resources so multiple pods can share the same GPU
+
+#### Runtime
+
+1. Add kubelet config parameter to point to external resource plugin
```

Comment:
Added a description of a new command line flag and the HTTP endpoints (assuming we stick with HTTP, which is up for debate).